From da3c9cd7a00be4ca031f33bec7264dbded7eccc8 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Mon, 31 Dec 2018 10:47:18 +0200
Subject: [PATCH] [SW-2751] preboot: print the SPL image type

The type is derived from the update procedure.
Types can be "candidate" or "stable".
During the update procedure, the new image type is "candidate".
Once the update procedure succeeds, meaning also the boot into
new image succeeds, the image is turned into "stable" type.

Change-Id: I3eb3905cbaa0093247a7f10731fa9cdb6ff279c7
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/preboot.c b/preboot.c
index ec294c0..c481acb 100644
--- a/preboot.c
+++ b/preboot.c
@@ -177,7 +177,7 @@
 	unsigned int image_base_offset;
 	unsigned int next_image, next_image_stable, next_image_modifiable;
 	bool try_new_image = false;
-	char *spl_image;
+	char *spl_image = "candidate";
 
 	modifiable_image_pending = readl(mmPSOC_GLOBAL_CONF_NON_RST_FLOPS_3);
 	if (modifiable_image_pending == 3) {
@@ -192,6 +192,7 @@
 		writel(0, mmPSOC_GLOBAL_CONF_NON_RST_FLOPS_3);
 
 	if (!try_new_image) {
+		spl_image = "stable";
 		next_image_stable = (*(unsigned int*)(HL_SPI_BASE + 0x60000) & 0xff) - '1';
 		if (next_image_stable < 2)
 			next_image = next_image_stable;
@@ -199,13 +200,10 @@
 			next_image = 0;
 	}
 
-	if (!next_image) { /* Primary */
+	if (!next_image)
 		image_base_offset = 0;
-		spl_image = "primary";
-	} else {/* Secondary */
+	else
 		image_base_offset = 0x800000;
-		spl_image = "secondary";
-	}
 
 	preboot_puts_raw("Preparing to run SPL ");
 	preboot_puts_raw(spl_image);
